astroquery/mast/core.py

```diff
@@ -261,6 +394,50 @@ def session_info(self, silent=None, verbose=None):
 
         return info_dict
 
+    def _fabric_request(self, method, url, params=None, data=None, headers=None,
+                 files=None, stream=False, auth=None, cache=False):
+        """
+        Override of the parent method:
+        A generic HTTP request method, similar to `~requests.Session.request`
+
+        This is a low-level method not generally intended for use by astroquery
+        end-users.
+
+        This method wraps the _request functionality to include raise_for_status
+        Caching is defaulted to False but may be modified as needed
+        Also parameters that allow for file download through this method are removed
+
+
+        Parameters
+        ----------
+        method : 'GET' or 'POST'
+        url : str
+        params : None or dict
+        data : None or dict
+        headers : None or dict
+        auth : None or dict
+        files : None or dict
+        stream : bool
+            See `~requests.request`
+        cache : bool
+            Default False. Use of bulit in _request caching
+
+        Returns
+        -------
+        response : `~requests.Response`
+            The response from the server.
+        """
+
+        start_time = time.time()
+
+        response = super(MastClass, self)._request(method, url, params=params, data=data, headers=headers,
+                                                   files=files, cache=cache, stream=stream, auth=auth)
```

Comment:
why is this using the parent class method?  Is there something about the internal method we don't want to use?

Reply:
We're not using the parent method here as we are in the process of refactoring our current use of `_request`. In the future we plan on renaming the current class `_request` to `_mashup_request` to keep mashup specific request items separate and allowing future API (most likely to MAST fabric services) methods to use the simpler `_fabric_request`